File: pset1/mario/more/mario.c
// Construct a pyramid using user height.
#include <stdio.h>
#include <cs50.h>

void space(int n);
void hashes(int n);

int main(void)
{

    int height;
    // Get the height from user.
    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);

    for (int i = 0; i <= height; i++)
    {
        // Print empty space before left side of pyramid.
        space(height-i);
        // Print hashes for the left half.
        hashes(i);
        // Print 2 empty space between both halves.
        space(2);
        // Print hashes for the right half.
        hashes(i);
        printf("\n");
    }
}

// Print empty space using value n.
void space(int n)
{
    for (int i = 0; i < n; i++) printf(" ");
}

// Print # using value n.
void hashes(int n)
{
    for (int i = 0; i < n; i++) printf("#");
}

File: pset6/mario/mario.py
# Port of mario.c in Python
import cs50

# get user input and ensure that it's between 0 and 8
while True:
    print("Height: ", end="")
    height = cs50.get_int()
    if (height > 0 and height < 9):
        break

# set str variables for empty space and block
space = " "
block = "#"

# iterate over each line in pyramid
for i in range(height):
    print((height-i-1)*space + (1+i)*block + 2*space + (1+i)*block)